src/DataSource/DibiFluentDataSource.php

```diff
@@ -35,10 +35,10 @@ class DibiFluentDataSource extends FilterableDataSource implements IDataSource,
 
 
 	/**
-	 * @param DibiFluent $data_source
+	 * @param DibiFluent|Dibi\Fluent $data_source
 	 * @param string $primary_key
 	 */
-	public function __construct(DibiFluent $data_source, $primary_key)
+	public function __construct($data_source, $primary_key)
```

Comment:
Can you please add an argument type check and in case of wrong argument given throw a `\InvalidArgumentException`? Thanks.